### PR TITLE
Support microservice with pre-built docker image and update secret 

### PR DESCRIPTION
**Pull request by carmenlau**
- Update microservice spec to support per-built docker image
- Update environment variable and secrets in microservice spec
- Update secret with type

### CODE REVIEW

File: features/305-microservice/microservice.md [hunk 1]
```diff
@@ -61,6 +82,20 @@ When the user run `skycli app deploy`, the following steps are taken:
...
+```yaml
+deployments:
+  my-awesome-service:
+    type: http-service
+    ....
+    environment:
+    - name: "DEMO_ENV"
+      value: "value of environment"
+      name: "MONGODB_URL"
+      secret: "<secret_name>"
```
Comment (chpapa) [inline features/305-microservice/microservice.md]: An idea, would it be better if have a shorthand when user want to use a secret name directly as an env var name too? Just to save a few keystrokes, like:

`secret2env: "<secret_name>"` or a `secret` without a `name`

is identical to

```
name: "<secret_name>"
secret: "<secret_name>"
```

Comment (louischan-oursky) [reply features/305-microservice/microservice.md]: `secret` without a `name` is unambiguous.

- `ENV_NAME: "plain value"`
- `ENV_NAME: "<secret_name>"`

is too ambiguous

File: features/270-next-skycli/commands.md [hunk 2]
```diff
@@ -376,10 +376,18 @@ aws_access_secret       2019-01-31T15:00:00+08:00
...
 
 Create new secret in app
 
+#### Flags
+
+- `--type` Secret type, used to facilitate programmatic handling of secret data. Available types: `opaque`, `dockerconfigjson`. Default is `opaque`.
+    - `opaque`: Secret used in environment variable.
+    - `dockerconfigjson`: Docker config used as image pull secret.
+
+- `--file` Secret value from file.
```
Comment (louischan-oursky) [inline features/270-next-skycli/commands.md]: Is `--file` specific to `--type=dockerconfigjson` or is it applicable to `--type=opaque` too, just like k8s?

Comment (carmenlau) [reply features/270-next-skycli/commands.md]: I think file should be independent to type, both type can use file.

File: features/305-microservice/microservice.md (hunk 1 above)
Comment (carmenlau) [reply features/305-microservice/microservice.md]: I think we can do `secret` without a `name`, but I will still prefer user provide the `name`  maybe in the guide so user will know they can specify the env name in `skygear.yaml`.

**Comment by louischan-oursky** [comment]
@kiootic Any comments?

**Pull request merged**